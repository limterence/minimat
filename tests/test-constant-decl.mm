/* test-constant-decl: definitions and scope */
constant int j      = 2 - 3;          /* definitions: all types */
constant float k    = -2.3;
constant bool l     = !true;
constant string s   = "hello, world";
constant matrix a   = [1.0; 2.0;]';
constant sequence x = new sequence(2);

void printconstants() {               /* global scope */
  printint(j);
  printfloat(k);
  printbool(l);
  printstring(s);
  println();
  printmat(a);
  printseq(x);
}

void printlocals() {    /* scope of constants suspended by local decls */
  int j;
  float k;
  bool l;
  string s;
  matrix a;
  int m;
  j = 99;
  k = 999.0;
  l = true;
  s = "goodbye";
  a = [99.9, 999.99, 9999.0;];
  m = 9999;
  printint(j);
  printfloat(k);
  printbool(l);
  printstring(s);
  println();
  printmat(a);
  printseq(x);
}

int main() {
  printconstants();
  printlocals();
}
