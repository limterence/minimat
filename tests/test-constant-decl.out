-1  -2.30 0 hello, world 
[1 x 2 float]
  1.00   2.00 
[2 int]
0 0 
99 999.00 1 goodbye 
[1 x 3 float]
 99.90 999.99 9999.00 
[2 int]
0 0 
